torch/csrc/autograd/variable.h

```diff
@@ -256,6 +263,35 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
   }
 };
 
+struct TORCH_API ViewInfo {
+  /// The base `Variable`
+  /// This cannot be a view of the same type as this one.
```

Comment:
It's not clear to the reader what "view of the same type" means. "Aren't you just meaning, the base can't itself be a view?" (And that gets into the discussion about forward/backward views being distinct, which needs a Note...)

Reply:
Yes I can rephrase that to say that it cannot be a forward view if this ViewInfo tracks a forward view and similarly for backward views.